luminance-gl/src/gl33/state.rs

```diff
@@ -470,14 +470,17 @@ impl GLState {
   }
 
   pub(crate) unsafe fn set_texture_unit(&mut self, unit: u32) {
-    if self.current_texture_unit != unit {
-      gl::ActiveTexture(gl::TEXTURE0 + unit as GLenum);
-      self.current_texture_unit = unit;
+    let unit = unit as GLenum;
+    if self.current_texture_unit.is_invalid(&unit) {
+      gl::ActiveTexture(gl::TEXTURE0 + unit);
+      self.current_texture_unit.set(unit);
     }
   }
 
   pub(crate) unsafe fn bind_texture(&mut self, target: GLenum, handle: GLuint) {
-    let unit = self.current_texture_unit as usize;
+    // Unwrap should be safe here, because we should always bind the texture unit before we bind the texture.
+    // Maybe this should be handled differently?
+    let unit = self.current_texture_unit.0.unwrap() as usize;
```

Comment:
I'm using unwrap here, as explained in the comment. Is there a better way for handling this?

Reply:
I think it’s okay. If such code fires, then it’s a bug in `luminance` — misuse of the `unsafe` layer, and it should panic, not get ignored. I’m fine with this. :)